### ISSUE TITLE
Q: do we have a target first localization language for ArcLight?

### ISSUE THREAD

**Issue by camillevilla** [projectblacklight/arclight#594]

**Comment by anarchivist** [comment]
@noahgh221 @seanaery  I see Duke had some interest. Do you have a specific language that you want to target?

**Comment by seanaery** [comment]
Looking at our past year of finding aids Analytics data, here are the most common languages set in our users' browsers:

Language | Sessions | Pct
-- | -- | --
English | 137,053 | 92.72%
Spanish | 1,609 | 1.09%
Chinese | 1,365 | 0.92%
French | 1,228 | 0.83%
German | 1,160 | 0.78%
Russian | 851 | 0.58%
Portuguese | 713 | 0.48%
Italian | 709 | 0.48%
Japanese | 496 | 0.34%
Dutch | 414 | 0.28%
Other | 2,218 | 1.50%
TOTAL | 147,816 | 100.00%

@noahgh221 Anything in particular in mind?

**Comment by seanaery** [comment]
Reconciling our ten most-used languages with the list of ten languages with [translations available upstream in Blacklight](https://github.com/projectblacklight/blacklight/tree/master/config/locales), it seems only Russian (6th) & Japanese (9th) are unique to us (Hungarian & Albanian in their place). I think we should target a language where those upstream translations exist.

**Comment by noahgh221** [comment]
Agreed. I think my interest in localization was not for any specific language, but just the idea of supporting them generally. Given our collection strengths, I would guess that Spanish and Chinese are the two most common languages (other than English) used in our finding aid corpus.

**Comment by anarchivist** [comment]
OK, thanks both!

@camillevilla What other information do you need to move this forward?

**Comment by camillevilla** [comment]
Thanks for this analysis @seanaery @noahgh221 @anarchivist. 

As a project, I think we need to make a decision re: how we want to accept translations from contributors. Once we choose an approach, I would be happy to close this ticket. This would be an interesting question to bring to PO meeting. I'd also invite @ggeisler's opinion on this, since he worked closely with the Italian and Chinese translators for Spotlight.

---
## Proposal A: Accept Partial / “In progress” languages into master branch
This is the approach we took with Spotlight due to the sheer number of translations that must be provided. To move the project forward, we chose to merge the minimum number of translations needed for visitors to interact with the site. At the time, we could not source translations for the entirety of the curator-facing UI. In that project [we do not fail the build for missing translations](https://github.com/projectblacklight/spotlight/blob/81543ee2c5f4dca941887bda87f809c4929ac47f/spec/i18n_spec.rb#L11-L14).

One downside to this approach is that it may not be clear to potential adopters that certain languages are "in progress" unless they spin up a local dev environment or compare translation file lengths in GitHub.

## Proposal B: Stage “in progress” languages in a feature branch
In this approach, we would accept translations as PRs to a feature branch marked "work in progress". This branch would not be merged into master until we have all keys.

---
Dealing with in-progress translations may not be a huge problem for ArcLight, given that it currently has a smaller number of keys.
- ArcLight: 68 lines
- Blacklight: 219 lines
- Spotlight: 781 lines

**Note:** As of #621, [missing translation keys in any language will fail the build](https://github.com/projectblacklight/arclight/pull/621/files#diff-57713778027b554e3084405e21cdf298).

**Comment by mejackreed** [comment]
Thanks for laying out the options @camillevilla .

I'd vote for Proposal A. Mostly for the reason that as the ArcLight project grows/changes, translations might also have to change or they may go missing. E.g We add a feature in the default language but don't have a translation readily available in the other language.

However, we may have to back off failing the build b/c of a missing translation to accommodate this approach.

**Comment by ggeisler** [comment]
I'd like to say Proposal B is preferable, because based on my experience with the Spotlight translation process, the hard part about getting translations for a new language wasn't completeness. The challenge is finding a willing translator that we trust has a) target language proficiency and b) enough understanding of user interfaces and the application to be translated to choose meaningful and appropriate translations. When we found translators with those qualities we had no problem getting them to provide all the needed translations.

However, I also had the same thought as @mejackreed above. Even when we get complete translations for the application at some point in time, as soon as we add new features that require additional translations, we have incompleteness. And finding a translator to provide these additional translations on short notice probably isn't something we can realistically expect (e.g., the two translators for the Chinese and Italian Spotlight translations are not part of the Blacklight community). And once a language is merged and public, I don't think we could pull it back out the application while we wait for translations for new feature that's been added to the application.

**Comment by anarchivist** [comment]
POs are in agreement that we should go with Proposal A. Thanks!

**Comment by mejackreed** [comment]
Closing this as the issue seems resolved.
